python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -202,3 +210,25 @@ def query_datasets(**kwargs):
         print("")
         table.pprint_all()
     print("")
+
+
+@click.command()
+@repo_argument(required=True)
+@dimensions_argument(help=unwrap("""DIMENSIONS are dimensions of the data IDs to yield. Will be expanded to a
+                                 complete DimensionGraph. --collections is required if using this
+                                 argument."""))
```

Comment:
`--collections` would be required for the `--datasets` option, not this argument.

I'd just change this docstring to "DIMENSIONS of the keys of the data IDs to yield.  Will be expanded to include any dependencies."  I think that's a bit more beginner-friendly.

Reply:
In fact I think it would help a lot if we gave some examples here and said "such as exposure, instrument, or tract".

Are we planning on at some point adding an option that would list the full record for the dimension? eg listing the exposure.obs_id or detector.full_name?

Reply:
> Are we planning on at some point adding an option that would list the full record for the dimension? eg listing the exposure.obs_id or detector.full_name?

There is a simpler Python API for that via a different registry method, `queryDimensionRecords`.  I figured we'd just make that another subcommand on another ticket; the full-records expanded data ID you can get back from `queryDataIds` (by calling `expanded()` on the result object) isn't something that lends itself to command-line pretty-printing.

Reply:
ok. That would be handy so I'll maybe take a look at that myself.